HW04/Kate Fedotova/arr.rb

```diff
@@ -0,0 +1,33 @@
+class Array
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |item_idx|
+      yield(self[item_idx])
+    end
+
+    self
+  end
+
+  def my_map
+    return clone.to_enum unless block_given?
```

Comment:
Why do you need `clone` here? Where did you find it? 🤔 😅
```suggestion
    return to_enum unless block_given?
```

Reply:
I found it in internet. Colne creates a copy of the array.